### PR TITLE
Updated documentation for LDataManager.

### LINKED ISSUE #1040: Get lag data notes into manual
@abarret I saw that you responded to an email on the mailing list with some notes on lag data (using `ibamr_lag_data.pdf`). Are these notes inside the manual? If not, would you please add them?

### PR DESCRIPTION
**Pull request by abarret**
This adds in notes I have for interacting with IB Lagrangian data. Fixes #1040.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Review by drwells — APPROVED** [review]
This looks fine to me but someone who has actually used this class should take a second look.

File: ibtk/include/ibtk/LDataManager.h [hunk 1]
```diff
@@ -86,7 +86,90 @@ namespace IBTK
  * for all inter-processor communications.  All access to instantiated
  * LDataManager objects is via the static method getManager().
  *
+ * Each Lagrangian point is associated with an LNode object which provides an interface between the Lagrangian and PETSc
+ * ordering as well as the data storage for forces and objects assosciated with the point. From each LNode object, we
```
Comment (boyceg) [inline ibtk/include/ibtk/LDataManager.h:90]: Do you mean something like:

```suggestion
 * ordering as well as the data storage for force specification objects and other objects associated with the point. From each LNode object, we
```

Resultant forces are stored in PETSc `Vec`s.

File: ibtk/include/ibtk/LDataManager.h [hunk 2]
```diff
@@ -86,7 +86,90 @@ namespace IBTK
...
+ * can retrieve the Lagrangian index as well as any associated node data. The LNode objects for each processor are
+ * contained in an LMesh object. The LMesh object contains two vector of LNodes: one containing just the local LNode
```
Comment (boyceg) [inline ibtk/include/ibtk/LDataManager.h:92]:
```suggestion
 * contained in an LMesh object. The LMesh object contains two vectors of LNodes: one containing just the local LNode
```

File: ibtk/include/ibtk/LDataManager.h [hunk 3]
```diff
@@ -86,7 +86,90 @@ namespace IBTK
...
+ * objects, and another containing 'ghost' LNodes. The 'ghost' LNodes are Lagrangian points located on other processors
```
Comment (boyceg) [inline ibtk/include/ibtk/LDataManager.h:93]:
```suggestion
 * objects, and another containing 'ghost' LNodes. The 'ghost' LNodes are Lagrangian points assigned to other processors
```
?

Comment (boyceg) [reply ibtk/include/ibtk/LDataManager.h:93]: (If we don't already do so, maybe we also need to talk somewhere about how nodes are currently assigned to processors.)

**Review by boyceg — COMMENTED** [review]
Looks good. Suggesting a few minor clarifications.

File: ibtk/include/ibtk/LDataManager.h (hunk 3 above)
Comment (abarret) [reply ibtk/include/ibtk/LDataManager.h:93]: Do you want to say something more than "Nodes are assigned to processors based on the distribution of patches"?

Comment (boyceg) [reply ibtk/include/ibtk/LDataManager.h]: I think that is about it. Could maybe say "currently based on"...?

**Pull request merged**